kreuzberg/_extractors/_pdf.py

```diff
@@ -163,6 +201,89 @@ def _validate_extracted_text(self, text: str, corruption_threshold: float = 0.05
 
         return (len(corruption_matches) / len(text)) < corruption_threshold
 
+    async def _extract_images_from_playa(self, doc: Document) -> list[ExtractedImage]:
+        async def extract_single_image(page_num: int, img_index: int, img_obj: Any) -> ExtractedImage | None:
```

Comment:
Don't online this inside another function. Also, refactor this - use chain and comorehensions